### PR TITLE
new optional state to add AppArmor rules, refs #6

### PR DESCRIPTION
**Pull request by daks**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

### What type of PR is this?

New feature

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

#6 

### Describe the changes you're proposing

New apparmor state, optional, to insert AppArmor rules to allow reading of the secrets dropin directory.
The state must be called directly or activated with a specific pillar.

This state has only been tested on Debian and may not be functional with other platforms without further work.

### Pillar / config required to test the proposed changes

```
strongswan:
 lookup:
   apparmor:
     add_rules: true
```

### Debug log showing how the proposed changes work

```
       ----------
                 ID: apparmor_usr.lib.ipsec.charon
           Function: file.managed
               Name: /etc/apparmor.d/local/usr.lib.ipsec.charon
             Result: True
            Comment: File /etc/apparmor.d/local/usr.lib.ipsec.charon updated
            Started: 12:22:55.046559
           Duration: 4.721 ms
            Changes:
              ----------
              diff:
                  New file
              mode:
                  0644
```

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

No test has been included because the state only works on Debian.
Testing would require creating a new test suite to only test this specific state on Debian

### Additional context

none

### CODE REVIEW

File: strongswan/apparmor.sls [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+# manage AppArmor rules for Strongswan
+# for the 'dropin' directories
+
+{%- set topdir = tpldir.split('/')[0] %}
+{%- from tpldir ~ "/map.jinja" import strongswan with context %}
+{%- from tpldir ~ "/libtofs.jinja" import files_switch with context %}
+
+apparmor_usr.lib.ipsec.charon:
```
Comment (vveliev-tc) [inline strongswan/apparmor.sls]:
```suggestion
strongswan-apparmor-file-apparmor-rule-file:
```

**Review by vveliev-tc — COMMENTED** [review]
would this apply to this formula?

File: strongswan/apparmor.sls [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - user: root
+    - group: root
+    - mode: '0644'
+    - template: jinja
+    - context:
+        secrets_dir: {{ strongswan.config.dropin_secrets }}
+    - watch_in:
+      - service: apparmor_service
+
+apparmor_service:
```
Comment (vveliev-tc) [inline strongswan/apparmor.sls]:
```suggestion
strongswan-apparmor-service-running-apparmor:
```

File: strongswan/apparmor.sls (hunk 1 above)
Comment (daks) [reply strongswan/apparmor.sls]: I am working on an additional commit to change all state ID. Is there any rule for the one you propose? I feel weird the repetition of 'apparmor' (the first one is the SLS name, what is the second one?) and 'file'.

Comment (daks) [reply strongswan/apparmor.sls]: Done in https://github.com/saltstack-formulas/strongswan-formula/pull/9/commits/0c3d1a8d7a77f3f7cd8445eb30112c4b8f53c9eb

**Pull request merged**